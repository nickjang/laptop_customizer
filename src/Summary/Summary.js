import React, { Component } from 'react';
import SummarySelected from '../SummarySelected/SummarySelected';
import Total from '../Total/Total';
import './Summary.css'

class Summary extends Component {
  static defaultProps = {
    selected: []
  };
  
  render() {
    const summary = Object.keys(this.props.selected).map((feature, idx) => {
      const featureHash = feature + '-' + idx;
      return <SummarySelected key={featureHash} feature={feature} idx={idx} selectedOption={this.props.selected[feature]} />;
    });

    return (
      <section className="main__summary">
        <h2>Your cart</h2>
        {summary}
        <Total selected={this.props.selected} />
      </section>
    );
  }
}

export default Summary;