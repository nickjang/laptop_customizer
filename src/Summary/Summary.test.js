import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Summary from './Summary';

const selected = {
  Processor: {
    name: '17th Generation Intel Core HB (7 Core with donut spare)',
    cost: 700
  },
  'Operating System': {
    name: 'Ubuntu Linux 16.04',
    cost: 200
  },
  'Video Card': {
    name: 'Toyota Corolla 1.5v',
    cost: 1150.98
  },
  Display: {
    name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
    cost: 1500
  }
}

const FEATURES = {
  Processor: [
    {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
    {
      name: 'Professor X AMD Fire Breather with sidewinder technology',
      cost: 1200
    }
  ],
  "Operating System": [
    {
      name: 'Ubuntu Linux 16.04',
      cost: 200
    },
    {
      name: 'Bodhi Linux',
      cost: 300
    }
  ],
  "Video Card": [
    {
      name: 'Toyota Corolla 1.5v',
      cost: 1150.98
    },
    {
      name: 'Mind mild breeze 2000',
      cost: 1345
    }
  ],
  Display: [
    {
      name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
      cost: 1500
    },
    {
      name: '15.3" HGTV (3840 x 2160) Home makeover edition',
      cost: 1400
    },
  ]
};

describe('<Summary />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Summary />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty when no props supplied', () => {
    const wrapper = shallow(<Summary />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders summary with selected options when props are supplied', () => {
    const wrapper = shallow(
      <Summary selected={selected} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});