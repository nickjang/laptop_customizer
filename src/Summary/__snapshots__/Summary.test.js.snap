// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> renders empty when no props supplied 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <Total
    selected={Array []}
  />
</section>
`;

exports[`<Summary /> renders summary with selected options when props are supplied 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <SummarySelected
    feature="Processor"
    idx={0}
    key="Processor-0"
    selectedOption={
      Object {
        "cost": 700,
        "name": "17th Generation Intel Core HB (7 Core with donut spare)",
      }
    }
  />
  <SummarySelected
    feature="Operating System"
    idx={1}
    key="Operating System-1"
    selectedOption={
      Object {
        "cost": 200,
        "name": "Ubuntu Linux 16.04",
      }
    }
  />
  <SummarySelected
    feature="Video Card"
    idx={2}
    key="Video Card-2"
    selectedOption={
      Object {
        "cost": 1150.98,
        "name": "Toyota Corolla 1.5v",
      }
    }
  />
  <SummarySelected
    feature="Display"
    idx={3}
    key="Display-3"
    selectedOption={
      Object {
        "cost": 1500,
        "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
      }
    }
  />
  <Total
    selected={
      Object {
        "Display": Object {
          "cost": 1500,
          "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
        },
        "Operating System": Object {
          "cost": 200,
          "name": "Ubuntu Linux 16.04",
        },
        "Processor": Object {
          "cost": 700,
          "name": "17th Generation Intel Core HB (7 Core with donut spare)",
        },
        "Video Card": Object {
          "cost": 1150.98,
          "name": "Toyota Corolla 1.5v",
        },
      }
    }
  />
</section>
`;
